### PR TITLE
fix incompatible file type checking on Windows

### LINKED ISSUE #62121: hostpath.type as `DirectoryOrCreate` does not work on windows
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature

**What happened**:
hostpath.type as `DirectoryOrCreate` does not work on windows, there will be mount failed error

```
---
apiVersion: v1
kind: Pod
metadata:
  name: aspnet-hostpath
spec:
  containers:
  - image: microsoft/aspnet
    name: aspnet-hostpath
    volumeMounts:
    - name: test-volume
      mountPath: '/azure'
  nodeSelector:
    beta.kubernetes.io/os: windows
  volumes:
  - name: test-volume
    hostPath:
      path: 'c:\var\lib\kublet\test'
      type: DirectoryOrCreate
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.11
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig storage
/sig windows
/assign

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #62121

**Special notes for your reviewer**:
/cc @msau42 @jsafrane 
/assign @andyzhangx 

**Release note**:

```release-note
fix incompatible file type checking on Windows
```

### CODE REVIEW

**Review by andyzhangx — APPROVED** [review]
/lgtm

File: pkg/util/mount/mount_windows.go [hunk 1]
```diff
@@ -211,18 +211,23 @@ func (mounter *Mounter) GetFileType(pathname string) (FileType, error) {
...
-		return FileTypeBlockDev, nil
-	case syscall.S_IFCHR:
-		return FileTypeCharDev, nil
-	case syscall.S_IFDIR:
+	// checks whether the mode is the target mode
+	isSpecificMode := func(mode, targetMode os.FileMode) bool {
+		return mode&targetMode == targetMode
+	}
+
+	mode := info.Mode()
```
Comment (msau42) [inline pkg/util/mount/mount_windows.go]: Can this be done for linux too?

Comment (dixudx) [reply pkg/util/mount/mount_windows.go]: Yes. Works for linux as well. Will create a common function for use.

**Comment by dixudx** [comment]
> exit status 1: mknod: '/tmp/test-get-filetype-289417242/test_char_dev': Operation not permitted

These tests got passed locally. Seems the CI is not friendly to such operations.
@msau42 Shall we omit such error?

**Comment by dixudx** [comment]
> Yeah, let's leave it out

@msau42 @andyzhangx Leave that error out. PTAL. Thanks.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Comment by andyzhangx** [comment]
hi @dixudx is it possible to also add nsenter_mount unit test for function like `PathIsDevice`? I just found a critcal bug in containerized kubelet which would cause UnmountDevice failed for all pods with volume mount,
 - related suspecious code is here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/mount/nsenter_mount.go#L223
 - related issue is here:
https://github.com/kubernetes/kubernetes/issues/62282

**Comment by dixudx** [comment]
@andyzhangx Have you tried the master branch? Or related to #61802?

**Pull request merged**